lib/auth/grpcserver.go

```diff
@@ -2356,7 +2356,9 @@ func userSingleUseCertsAuthChallenge(gctx *grpcContext, stream proto.AuthService
 	return mfaDev, nil
 }
 
-func userSingleUseCertsGenerate(ctx context.Context, actx *grpcContext, req proto.UserCertsRequest, mfaDev *types.MFADevice) (*proto.SingleUseUserCert, error) {
+// mfaVerifiedSessionExpires is the expiry of the user cert from which this single use cert is derived from.
+// See https://github.com/gravitational/teleport/issues/18544.
+func userSingleUseCertsGenerate(ctx context.Context, actx *grpcContext, req proto.UserCertsRequest, mfaDev *types.MFADevice, mfaVerifiedSessionExpires time.Time) (*proto.SingleUseUserCert, error) {
```

Comment:
```suggestion
func userSingleUseCertsGenerate(ctx context.Context, actx *grpcContext, req proto.UserCertsRequest, mfaDev *types.MFADevice) (*proto.SingleUseUserCert, error) {
```

You don't need to pass in the expiry as an argument, you already have it in `actx`.